core/src/main/java/io/jenkins/pluginhealth/scoring/probes/AbstractGitHubWorkflowProbe.java

```diff
@@ -0,0 +1,145 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is an abstract class that looks for desired configuration in the files present in GitHub Workflows directory.
+ *
+ * @return The class returns success when the configuration is found, otherwise returns a failure.
+ */
+
+public abstract class AbstractGitHubWorkflowProbe extends Probe {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractGitHubWorkflowProbe.class);
+    private static final String WORKFLOWS_DIRECTORY = ".github/workflows";
+
+    @Override
+    protected ProbeResult doApply(Plugin plugin, ProbeContext context) {
+        final Path repository = context.getScmRepository();
+        final Path workflowPath = repository.resolve(WORKFLOWS_DIRECTORY);
+
+        if (!Files.exists(workflowPath)) {
+            return ProbeResult.failure(key(), "Plugin has no GitHub Action configured");
+        }
+
+        try (Stream<Path> files = Files.find(workflowPath, 1, (path, $) -> Files.isRegularFile(path))) {
+            boolean isWorkflowConfigured = files
+                .map(file -> readWorkflowFile(file))
+                .filter(workflow -> isWorkflowJobsNullOrEmpty(workflow))
+                .flatMap(workflow -> workflow.jobs().values().stream())
+                .map(WorkflowJobDefinition::uses)
+                .filter(Objects::nonNull)
+                .anyMatch(jobDefinition -> jobDefinition.startsWith(getWorkflowDefinition()));
+
+            return isWorkflowConfigured ?
+                ProbeResult.success(key(), getSuccessMessage()) :
+                ProbeResult.failure(key(), getFailureMessage());
+        } catch (IOException e) {
+            LOGGER.warn("Couldn't not read file at {} on {}", key(), e);
+            return ProbeResult.error(key(), e.getMessage());
+        }
+    }
+
+    /**
+     * Returns the path to the GitHub Workflow definition which should be use in one of the actions of the plugin repository.
+     *
+     * @return the workflow definition used in one of the jobs of one of the actions of the plugin repository.
```

Comment:
Not exactly, this method returns the path of the workflow you are searching for in the actions defined in the plugin repository. Your sentence sounds like it is actually used, when you don't know yet.